### PR TITLE
arm: imx: add iMX7D PICO MBL board definition

### PR DESCRIPTION
**Pull request by niej**
This patch adds an OP-TEE port for the i.MX7D PICO for the MBED Linux OS
boot flow.

BootROM -> ATF/BL2
ATF     -> FIP {u-boot, OPTEE}
OPTEE   -> {populates DTB overlay}
u-boot  -> FIT {DTB, Kernel, initramfs}
           Merges DTB and OPTEE DTB-overlay
Linux

CFG_NS_ENTRY_ADDR = 0x87800000 is the entry point of u-boot
CFG_DT_OVERLAY = y adds DTB overlay fragments to the passed DTB

make CROSS_COMPILE=arm-linux-gnueabihf- PLATFORM=imx \
     PLATFORM_FLAVOR=mx7dpico_mbl ARCH=arm CFG_PAGEABLE_ADDR=0 \
     CFG_TEE_CORE_NB_CORE=2

Signed-off-by: Jun Nie <jun.nie@linaro.org>

### CODE REVIEW

**Comment by bryanodonoghue** [comment]
LTGM

File: core/arch/arm/plat-imx/conf.mk [hunk 1]
```diff
@@ -99,6 +100,17 @@ CFG_DDR_SIZE ?= 0x40000000
...
 endif
 
+ifneq (,$(filter $(PLATFORM_FLAVOR),mx7dpico_mbl))
+CFG_DDR_SIZE ?= 0x20000000
+CFG_NS_ENTRY_ADDR ?= 0x87800000
+CFG_DT_ADDR ?= 0x83100000
+CFG_UART_BASE ?= UART5_BASE
+CFG_BOOT_SECONDARY_REQUEST ?= n
+CFG_EXTERNAL_DTB_OVERLAY = y
+CFG_IMX_WDOG_EXT_RESET = y
```
Comment (jforissier) [inline core/arch/arm/plat-imx/conf.mk]: `?=` please

**Comment by jforissier** [comment]
One more (minor) comment: you could/should simplify the `make` command in the commit message. `make PLATFORM=imx-mx7dpico_mbl` should be enough AFAICT.

**Comment by MrVan** [comment]
With @jforissier 's comment resolved, `Reviewed-by: Peng Fan <peng.fan@nxp.com>`

**Comment by niej** [comment]
@jforissier @MrVan  Thanks for review! Patch is updated with addressing all comments.

**Comment by jforissier** [comment]
@niej thanks!

**Pull request merged**